pkg/ottl/functions.go

```diff
@@ -291,3 +375,27 @@ func buildSlice[T any](argVal value, argType reflect.Type, buildArg buildArgFunc
 
 	return vals, nil
 }
+
+func NewOptionalWithDefault[T any](def T) Optional[T] {
+	return Optional[T]{
+		Val: def,
+	}
+}
+
+type Optional[T any] struct {
```

Comment:
I agree that an interface would be nice, but as a convenience for function creator's we'll need a `StandardOptional` struct anyways.

Reply:
Could you elaborate? I'm having trouble using an interface right instead of a struct because the parser can't use the zero value for an interface and I can't figure out how to have the parser fix that on its own. We would need to have it create a `StandardOptional[T]` struct with the necessary data, then set it on the `Optional[T]` field in the `Arguments` struct. So for now we're stuck with just an `Optional` struct.

Reply:
I have no solution at the moment, only the theoretical concept that an interface would be great lol Is there any way to unexport the fields and only expose getting and setting them via functions that we execute with reflection? That would make the exported Optional type easier to modify in the feature without breaking changes to the API

Reply:
Unfortunately we can't call private methods with reflection, but we can work around the reflection limitations by having `Optional[T]` implement an interface that only deals with `any` and `reflect.Value` types. It's all a very delicate puzzle, and for now I think I've exhausted the options I'm aware of outside of meddling with `unsafe.Pointer` types.

Reply:
I'm still unable to think of a way to use an interface, but an `Optional` struct will probably meet all our needs. If you think of a reason why we will need an interface I will take another look at whether it's feasible.

Reply:
I'm ok moving forward with the Optional struct